src/plugins/orangehrmPerformancePlugin/Api/EmployeeTrackerAPI.php

```diff
@@ -0,0 +1,191 @@
+
+    public const FILTER_INCLUDE_EMPLOYEES = 'includeEmployees';
+    public const FILTER_NAME_OR_ID = 'nameOrId';
+
+    public const PARAM_RULE_FILTER_NAME_OR_ID_MAX_LENGTH = 100;
+
+    private ?PerformanceTrackerService $employeeTrackerService = null;
+
+    /**
+     * @return PerformanceTrackerService
+     */
+    public function getPerformanceTrackerService(): PerformanceTrackerService
+    {
+        if (!$this->employeeTrackerService instanceof PerformanceTrackerService) {
+            $this->employeeTrackerService = new PerformanceTrackerService();
+        }
+        return $this->employeeTrackerService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $employeeTrackerSearchFilterParams = $this->getEmployeeTrackerSearchFilterParams();
+        $this->setSortingAndPaginationParams($employeeTrackerSearchFilterParams);
+
+        $empNumber = $this->getRequestParams()->getIntOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            CommonParams::PARAMETER_EMP_NUMBER
+        );
+
+        if (!is_null($empNumber)) {
+            $employeeTrackerSearchFilterParams->setEmpNumbers([$empNumber]);
+        } else {
+            $accessibleEmpNumbers = $this->getUserRoleManager()->getAccessibleEntityIds(PerformanceTrackerReviewer::class);
+            $employeeTrackerSearchFilterParams->setEmpNumbers($accessibleEmpNumbers);
+        }
+
+        $employeeTrackerList = $this->getPerformanceTrackerService()
+            ->getPerformanceTrackerDao()
+            ->getEmployeeTrackerList($employeeTrackerSearchFilterParams);
+        $employeeTrackerCount = $this->getPerformanceTrackerService()
+            ->getPerformanceTrackerDao()
+            ->getEmployeeTrackerCount($employeeTrackerSearchFilterParams);
+
+        return new EndpointCollectionResult(
+            EmployeeTrackerModel::class,
+            $employeeTrackerList,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $employeeTrackerCount])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    CommonParams::PARAMETER_EMP_NUMBER,
+                    new Rule(Rules::CALLBACK, [
+                        function ($empNumber) {
+                            if (!(is_numeric($empNumber) && $empNumber > 0)) {
+                                return false;
+                            }
+
+                            return in_array(
+                                $empNumber,
+                                $this->getUserRoleManager()
+                                    ->getAccessibleEntityIds(PerformanceTrackerReviewer::class)
+                            );
+                        }
+                    ])
+                )
+            ),
```

Comment:
```suggestion
            $this->getValidationDecorator()->notRequiredParamRule(
                new ParamRule(
                    CommonParams::PARAMETER_EMP_NUMBER,
                    new Rule(Rules::POSITIVE),
                    new Rule(Rules::IN_ACCESSIBLE_ENTITY_ID, [PerformanceTrackerReviewer::class])
                )
            )
```

Reply:
I had to add `InAccessibleEntityIdOption` to make this work. please check if the approach is correct